arangod/Agency/Supervision.cpp

```diff
@@ -3596,7 +3602,8 @@ void Supervision::checkUndoLeaderChangeActions() {
         return result;
       }
 
-      auto logId = replication2::LogId::fromString(id);
+      // TODO: Check if this is correct
+      auto logId = replication2::LogId::fromString(id.c_str());
```

Comment:
If `id` is an `std::string`, no need to pass its `c_str()` value here.
```suggestion
      auto logId = replication2::LogId::fromString(id);
```

Reply:
Done